kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/schema/matchSchema.kt

```diff
@@ -92,9 +105,10 @@ private fun validate(
                "Expected items between ${expected.minItems} and ${expected.maxItems}, but was ${tree.elements.size}"
             )
             val matcherViolation = violation(expected.matcher, tree.elements.asSequence())
-            matcherViolation + sizeViolation + tree.elements.flatMapIndexed { i, node ->
-               validate("$currentPath[$i]", node, expected.elementType)
-            }
+            matcherViolation + sizeViolation +
+               (expected.elementType?.violation(tree)
+                  ?: expected.contains?.violation(tree)
```

Comment:
I think JSON schema definition  allows both elementType and contains to be specified in conjunction? For instance, the elementType might be object, and contains might be object with a certain property. Both can be true at the same time I think :)